lang/java/reef-io/src/main/java/org/apache/reef/io/network/shuffle/driver/ShuffleDriverImpl.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network.shuffle.driver;
+
+import org.apache.reef.driver.task.CompletedTask;
+import org.apache.reef.driver.task.FailedTask;
+import org.apache.reef.driver.task.RunningTask;
+import org.apache.reef.evaluator.context.parameters.ContextStartHandlers;
+import org.apache.reef.evaluator.context.parameters.ContextStopHandlers;
+import org.apache.reef.exception.evaluator.NetworkException;
+import org.apache.reef.io.network.NetworkServiceClient;
+import org.apache.reef.io.network.shuffle.ns.ShuffleControlLinkListener;
+import org.apache.reef.io.network.shuffle.ns.ShuffleControlMessageCodec;
+import org.apache.reef.io.network.shuffle.ns.ShuffleControlMessageHandler;
+import org.apache.reef.io.network.shuffle.params.SerializedShuffleSet;
+import org.apache.reef.io.network.shuffle.params.ShuffleControlMessageNSId;
+import org.apache.reef.io.network.shuffle.task.ShuffleClient;
+import org.apache.reef.io.network.shuffle.task.ShuffleContextStartHandler;
+import org.apache.reef.io.network.shuffle.task.ShuffleContextStopHandler;
+import org.apache.reef.io.network.shuffle.descriptor.ShuffleDescriptor;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Injector;
+import org.apache.reef.tang.JavaConfigurationBuilder;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+
+import javax.inject.Inject;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ *
+ */
+final class ShuffleDriverImpl implements ShuffleDriver {
+
+  private final Injector injector;
+  private final ConfigurationSerializer confSerializer;
+  private final ShuffleControlLinkListener linkListener;
+  private final ShuffleControlMessageHandler messageHandler;
+  private final ConcurrentMap<Class<? extends Name>, ShuffleManager> managerMap;
+
+  @Inject
+  public ShuffleDriverImpl(
+      final Injector injector,
+      final ConfigurationSerializer confSerializer,
+      final NetworkServiceClient nsClient,
```

Comment:
Could you merge with `apache/master` in order to use  `NetworkConnectionService`?